lib/auto-languageclient.ts

```diff
@@ -378,23 +392,31 @@ export default class AutoLanguageClient {
     );
   }
 
+  private handleCloseFailure(code: number | null, signal: NodeJS.Signals | null): void {
+    if (code !== 0 && signal === null) {
+      atom.notifications.addError(
+        `${this.getServerName()} language server for ${this.getLanguageName()} was closed with code: ${code}.`
```

Comment:
In my opinion, this is an error, but in the above, the `exit` is considered a logger.debug

Reply:
We can also add a `disconnect` listener.

Reply:
Ya I would think this should be an error by default as long as there is a way for packages to override it if it shouldn't be an error for them.

Reply:
They can do this by overriding the method.